sdcm/cluster_gce.py

```diff
@@ -197,64 +201,90 @@ def _get_persistent_disk_struct(self, name, disk_size, disk_type='pd-ssd', dc_id
         nodes = []
-        for node_index in range(self._node_index + 1, self._node_index + count + 1):
-            name = "%s-%s-%s" % (self.node_prefix, dc_idx, node_index)
-            gce_disk_struct = list()
-            gce_disk_struct.append(self._get_root_disk_struct(name=name,
-                                                              disk_type=self._gce_image_type,
-                                                              dc_idx=dc_idx))
-            for i in range(self._gce_n_local_ssd):
-                gce_disk_struct.append(self._get_local_ssd_disk_struct(name=name, index=i, dc_idx=dc_idx))
-            if self._add_disks:
-                for disk_type, disk_size in self._add_disks.iteritems():
-                    disk_size = int(disk_size)
-                    if disk_size:
-                        gce_disk_struct.append(self._get_persistent_disk_struct(name=name, disk_size=disk_size,
-                                                                                disk_type=disk_type, dc_idx=dc_idx))
-            self.log.info(gce_disk_struct)
-            # Name must start with a lowercase letter followed by up to 63
-            # lowercase letters, numbers, or hyphens, and cannot end with a hyphen
-            assert len(name) <= 63, "Max length of instance name is 63"
-            instance = self._gce_services[dc_idx].create_node(name=name,
-                                                              size=self._gce_instance_type,
-                                                              image=self._gce_image,
-                                                              ex_network=self._gce_network,
-                                                              ex_disks_gce_struct=gce_disk_struct)
-            self.log.info('Created instance %s', instance)
-            cluster.GCE_INSTANCES.append(instance)
-            try:
-                n = GCENode(gce_instance=instance,
-                            gce_service=self._gce_services[dc_idx],
-                            credentials=self._credentials[0],
-                            gce_image_username=self._gce_image_username,
-                            node_prefix=self.node_prefix,
-                            node_index=node_index,
-                            base_logdir=self.logdir,
-                            dc_idx=dc_idx)
-                nodes.append(n)
-            except Exception as ex:
-                self.log.exception('Failed to create node: %s', ex)
-            else:
-                self._node_index += 1
-                self.nodes += [n]
-
-                local_nodes = [n for n in self.nodes if n.dc_idx == dc_idx]
-                if len(local_nodes) > len(nodes):
-                    n.is_addition = True
-
-        assert len(nodes) == count, 'Fail to create {} instances'.format(count)
-        self.log.info('added nodes: %s', nodes)
+        if cluster.Setup.REUSE_CLUSTER:
+            instances = self._get_instances()
```

Comment:
I think cluster Base class should have _instances_ property when the instances are stored